### DISCUSSION TITLE
Setseed doesn't affect `... using sample N`

### DISCUSSION THREAD

**Discussion by ljmartin** [duckdb/duckdb#14326] [Q&A] [answered]
### What happens?

Hi DDB, 

I'm looking to randomly select from a table or parquet, while setting the random seed so that the selection is the same each time. For example:
```
select setseed(0.1);

CREATE TABLE numbers AS SELECT * FROM generate_series(1, 50) AS t(number);

select * from numbers using sample 5;
```
running this a few times yields different results on each re-run. The `setseed` function does work for generating RANDOM() numbers. 

Is it also applicable in any way to random sampling?

Thanks!

### To Reproduce

```
select setseed(0.1);

CREATE TABLE numbers AS SELECT * FROM generate_series(1, 50) AS t(number);

select * from numbers using sample 5;
```

### OS:

OSX Sonoma 14.4

### DuckDB Version:

v1.1.1 af39bd0dcf

### DuckDB Client:

CLI

### Hardware:

N/A, Apple M1 Max

### Full Name:

Lewis Martin

### Affiliation:

OpenBench

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by ljmartin** [comment, marked as answer]
Turns out this is all explained in the docs. The setseed docs specify that it is only for the 'random' function. The example page https://duckdb.org/docs/sql/samples.html shows you can set the seed like:
```
 select setseed(0.1);

CREATE TABLE numbers AS SELECT * FROM generate_series(1, 50) AS t(number);

select * from numbers using sample reservoir(5 rows) repeatable (100);
```

**Reply by ChandlerLutz** [reply]
Is it possible to do something like this with a `GROUP BY` statement to mimic stratified random sampling? I couldn't figure it out from the docs.
